contentlayer.config.ts

```diff
@@ -363,6 +378,31 @@ export const Transcript = defineDocumentType(() => ({
     source_file: { type: "string" },
   },
   computedFields: {
+    tagsDetailed: {
+      type: "list",
+      resolve: (doc) => {
+        const topicsMap = doc?.tags as any || [];
```

Comment:
something like the code below is easier to follow.
Are there instances where tags are not arrays?
```suggestion
        const topics: string[] = [];
        if (Array.isArray(doc?.tags?._array) {
          topics.push(doc.tags._array)
        }
```

Reply:
Yh, because of how content layer does the doc, docs.tags isn't actually an array till you do _array, and for typing we typed tags as an array which we do we need. 

Below is what we get with ```doc?.tags```
<img width="228" alt="Screenshot 2024-12-20 at 10 37 42" src="https://github.com/user-attachments/assets/de7d566a-a6ba-4b28-8e5b-7bc85f0dfedd" />

I think I could name it better and put a comment there